source-academy/cadet-frontend
# Mobile Workspace v1

### Description

This PR added mobile friendliness to the current source academy playground workspace.
An mobile workspace is created, pulling different existing components together for a more mobile friendly UI. 
This mobile workspace will be render whenever screen width is smaller than the conventional breakpoint 768px.

We introduced `react-responsive` and `react-draggable` dependencies.

Autorun and Execution time settings have been intentionally left out of the mobile ControlBar.

### Type of change

### Changes
You can go to this [_**staging deployment site**_](https://sa-mobile.vercel.app/playground) to see the app in action! (please resize your browser to mobile screen resolutions or open this site on you phone)
Here are some screen shots.
<div style="display:flex">
<img src="https://user-images.githubusercontent.com/39845424/106909185-fe786800-673a-11eb-9045-1ac0c344cc4c.png" height="440">
<img src="https://user-images.githubusercontent.com/39845424/106910384-21efe280-673c-11eb-93e1-e793306c7b70.png" height="440">
<img src="https://user-images.githubusercontent.com/39845424/106910777-86ab3d00-673c-11eb-9120-c774fd88b5ea.png" height="440">
</div>

### How to test

This project mainly has UI changes and manual testing should be done to ensure features work as intended across different devices.

Please refer to this wiki page on [_**mobile workspace development**_](https://github.com/source-academy/cadet-frontend/wiki/Mobile-Development-Guide#manual-browser-resizing-checks) for more detail of testing and feature description. These tests focus on the responsive features of the app (i.e. how the app handles resizing)

However, please note that DO NOT **"add to home screen"** in your mobile devices yet, as currently the progressive web app is not configured.

Over and above, here is a non-exhaustive list of features that are to be tested out for this PR (please think of possible ways to introduce bugs - e.g. large programs):
- Running the Stepper tab (both breakpoint and normal program execution)
- Data Visualizer (and overflow)
- Environment Visualizer (and overflow)
- Inspector
- Runes
- Editor breakpoints
- Toggling to other libraries
- Resizing desktop SideContentTabs and then toggling to the mobile Playground

This is still an on-going project so if you see any bug or UI imperfection please file a issue to us. (@YaleChen299 and @chownces)
To other frontend teams: we really appreciate if you could help us in testing your components in the mobile setting and contact us if you have any issue. 

### Checklist
